aggregates/src/androidMain/kotlin/org/dhis2/mobile/aggregates/data/DataSetInstanceRepositoryImpl.kt

```diff
@@ -336,6 +336,49 @@ internal class DataSetInstanceRepositoryImpl(
             }
     }
 
+    override suspend fun getLegend(
```

Comment:
could we retrieve this information when we get the Dataelement the first time?

Reply:
The legend is being retrieve when fetching the value of the dataelement yes. And it is also requested when the value changes.